src/lib/icli/index.js

```diff
@@ -0,0 +1,354 @@
+        }
+      });
+      if (errorMessages.length > 0) {
+        return errorMessages;
+      }
+      return true;
+    };
+  }
+};
+
+module.exports = icli;
+
+
+/**
+ * Calculate parameter names and add a "validate" property if needed
+ * Return the list of commander arguments and the list of commander options
+ * @param  {Array} parameters - a list of parameter configurations
+ * @return {Object} - an object containing the list of arguments and the list of options
+ */
+function parseParameters(parameters) {
+  const args = [];
+  const options = [];
+
+  // Extract options and arguments
+  // Enrich parameter configs with a name calculated from "cmdSpec"
+  // Create automatic validators
+  _.forEach(parameters, parameter => {
+    if (_.startsWith(parameter.cmdSpec, '-')) {
+      // Case the parameter is an option
+      options.push(parseOptionSpec(parameter));
+    } else {
+      // case the parameter is an argument
+      args.push(parseArgumentSpec(parameter));
+    }
+
+    // Automaticaly add validators
+    // If the parameter configuration already has a validator, we do not override it
+    if (!parameter.validate) {
+      // We create validators for all "list" and "checkbox" parameters
+      if (['list', 'checkbox'].indexOf(parameter.type) !== -1) {
+        // We automatically add a validator to list and checkbox parameters
+        parameter.validate = icli.generateListValidation(parameter.choices, parameter.validationMsgLabel);
+      }
+    }
+  });
+
+  return { args, options };
+}
+
+/**
+ * Parse a parameter of type "option" to calculate the name and generate the commander option
+ * @param  {Object} parameter - a parameter definition that will be enriched with a correct name
+ * @return {Object} - the properties of the commander option
+ */
+function parseOptionSpec(parameter) {
+  // @see https://github.com/tj/commander.js/blob/33751b444a578259a7e37a0971d757452de3f228/index.js#L44-L46
+  const flags = parameter.cmdSpec.split(/[ ,|]+/);
+  if (flags.length > 1 && !/^[[<]/.test(flags[1])) { flags.shift(); }
+  parameter.name = _.camelCase(flags.shift());
+  return {
+    flags: parameter.cmdSpec,
+    description: parameter.description,
+    coercion: parameter.coercion || getCoercionForType(parameter.type)
+  };
+}
+
+/**
+ * Parse a parameter of type "argument" to calculate the name and generate the commander option
+ * @param  {Object} parameter - a parameter definition that will be enriched with a correct name
+ * @return {Object} - a commander argument
+ */
+function parseArgumentSpec(parameter) {
+  parameter.name = _.camelCase(parameter.cmdSpec);
+  return parameter.cmdSpec;
+}
+
+/**
+ * Get default coercion for a parameter type
+ * @param  {string} type - a parameter type
+ * @return {function|null} - the coercion function to applu to the command line argument/option
+ */
+function getCoercionForType(type) {
+  switch(type) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Keyword "switch" must be followed by whitespace.](https://www.codacy.com/app/lager/lager/file/2779185918/issues/source?bid=3360491&fileBranchId=3365065#l193)